src/parser/hunter/beastmastery/modules/talents/AnimalCompanion.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import StatisticBox from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import { formatNumber } from 'common/format';
+import PackAlpha from 'parser/hunter/beastmastery/modules/spells/azeritetraits/PackAlpha';
+
+/**
+ * Your Call Pet additionally summons the first pet from your stable.
+ * This pet will obey your Kill Command, but cannot use pet family abilities.
+ *
+ * Additionally this talent baseline reduces all pet damage by 40%.
+ *
+ * Example log: https://www.warcraftlogs.com/reports/aNRJKvknX1FYqQ6G#fight=4&type=damage-done&source=7
+ */
+
+class AnimalCompanion extends Analyzer {
+  static dependencies = {
+    packAlpha: PackAlpha,
+  };
+  damage = 0;
+  pets = [];
+  mainPetName = null;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.ANIMAL_COMPANION_TALENT.id);
+  }
+
+  on_byPlayerPet_damage(event) {
+    const sourcePet = this.owner.playerPets.find(pet => pet.id === event.sourceID);
+    const foundPet = this.pets.find(pet => pet.sourceID === event.sourceID);
+    const damage = event.amount + (event.absorbed || 0);
+    if (!foundPet) {
+      this.pets.push({ petName: sourcePet.name, sourceID: event.sourceID, damage: damage });
+    } else {
+      const indexOfPet = this.pets.findIndex(index => index.sourceID === event.sourceID);
+      this.pets[indexOfPet].damage += damage;
+    }
+  }
+
+  on_finished() {
+    let max = -Infinity;
+    let x;
+    for (x in this.pets) {
```

Comment:
I think this could also be simplified with (haven't tried)
```javascript
this.pets.forEach(pet => {
  if (pet.damage > max) {
    max = pet.damage;
    this.mainPetName = pet.petName;
  }
});
```